[2025-04-26T23:19:33.369+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: earth_monitoring_system.fetch_air_quality_data manual__2025-04-26T23:19:30.769066+00:00 [queued]>
[2025-04-26T23:19:33.380+0000] {taskinstance.py:1103} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: earth_monitoring_system.fetch_air_quality_data manual__2025-04-26T23:19:30.769066+00:00 [queued]>
[2025-04-26T23:19:33.380+0000] {taskinstance.py:1308} INFO - Starting attempt 1 of 2
[2025-04-26T23:19:33.406+0000] {taskinstance.py:1327} INFO - Executing <Task(BashOperator): fetch_air_quality_data> on 2025-04-26 23:19:30.769066+00:00
[2025-04-26T23:19:33.410+0000] {standard_task_runner.py:57} INFO - Started process 1368 to run task
[2025-04-26T23:19:33.413+0000] {standard_task_runner.py:84} INFO - Running: ['airflow', 'tasks', 'run', 'earth_monitoring_system', 'fetch_air_quality_data', 'manual__2025-04-26T23:19:30.769066+00:00', '--job-id', '5', '--raw', '--subdir', 'DAGS_FOLDER/earth_monitoring_dag.py', '--cfg-path', '/tmp/tmpq8htcf2r']
[2025-04-26T23:19:33.415+0000] {standard_task_runner.py:85} INFO - Job 5: Subtask fetch_air_quality_data
[2025-04-26T23:19:33.462+0000] {logging_mixin.py:150} WARNING - /home/airflow/.local/lib/python3.7/site-packages/airflow/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2025-04-26T23:19:33.574+0000] {task_command.py:410} INFO - Running <TaskInstance: earth_monitoring_system.fetch_air_quality_data manual__2025-04-26T23:19:30.769066+00:00 [running]> on host d522fce2db24
[2025-04-26T23:19:33.820+0000] {taskinstance.py:1547} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='earth_monitoring_system' AIRFLOW_CTX_TASK_ID='fetch_air_quality_data' AIRFLOW_CTX_EXECUTION_DATE='2025-04-26T23:19:30.769066+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-04-26T23:19:30.769066+00:00'
[2025-04-26T23:19:33.821+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-04-26T23:19:33.822+0000] {subprocess.py:75} INFO - Running command: ['/bin/bash', '-c', 'cd /opt/airflow/data_collection && python fetch_openaq.py']
[2025-04-26T23:19:33.828+0000] {subprocess.py:86} INFO - Output:
[2025-04-26T23:19:34.354+0000] {subprocess.py:93} INFO - 2025-04-26 23:19:34,351 - WARNING - No API key found. Set OPENAQ_API_KEY in .env file.
[2025-04-26T23:19:34.356+0000] {subprocess.py:93} INFO - 2025-04-26 23:19:34,353 - INFO - Starting data collection from OpenAQ v3
[2025-04-26T23:19:34.357+0000] {subprocess.py:93} INFO - 2025-04-26 23:19:34,354 - INFO - Fetching data from OpenAQ API v3 with limit 1000
[2025-04-26T23:19:34.791+0000] {subprocess.py:93} INFO - 2025-04-26 23:19:34,790 - ERROR - Error fetching data from OpenAQ: 401 Client Error: Unauthorized for url: https://api.openaq.org/v3/measurements?limit=1000&has_geo=true
[2025-04-26T23:19:34.794+0000] {subprocess.py:93} INFO - 2025-04-26 23:19:34,792 - WARNING - No data received from API
[2025-04-26T23:19:34.795+0000] {subprocess.py:93} INFO - 2025-04-26 23:19:34,792 - INFO - Data collection completed
[2025-04-26T23:19:34.808+0000] {subprocess.py:97} INFO - Command exited with return code 0
[2025-04-26T23:19:34.912+0000] {taskinstance.py:1350} INFO - Marking task as SUCCESS. dag_id=earth_monitoring_system, task_id=fetch_air_quality_data, execution_date=20250426T231930, start_date=20250426T231933, end_date=20250426T231934
[2025-04-26T23:19:34.984+0000] {local_task_job_runner.py:225} INFO - Task exited with return code 0
[2025-04-26T23:19:35.047+0000] {taskinstance.py:2653} INFO - 0 downstream tasks scheduled from follow-on schedule check
